### DISCUSSION TITLE
Update Developer Badges

### DISCUSSION THREAD

**Discussion by televandalist** [RetroAchievements/RAWeb#775] [Ideas]
Achievement developers get two badges that 'evolve' over time: one for how many times their achievements have been earned and one for how many points have been awarded by their achievements. These stats stop at 'awarded 6000 times' and 'awarded 70000 points' - considering the current records for single developers are 632,915 achievements awarded and 6,925,068 points awarded, I feel that these badges could use an update.

I'm not sure where all the necessary code is, but bootstrap.php has the following code in lines 61-82:

$developerCountBoundaries = [5, 10, 50, 100, 200, 400, 600, 800, 1000, 2000, 3000, 4000, 5000, 6000];
$developerPointBoundaries = [
    100,
    200,
    300,
    500,
    800,
    1000,
    1500,
    2000,
    3000,
    4000,
    5000,
    10000,
    15000,
    20000,
    30000,
    40000,
    50000,
    60000,
    70000,
];

The milestones and badge images would obviously need to be changed as well...

**Comment by televandalist** [comment]
_considering the current records for single developers are 632,915 achievements awarded and 6,925,068 points awarded, I feel that these badges could use an update._

This is now 936,301 achievements awarded and 10,214,256 points awarded, for Brian and dude, respectively.

**Comment by Tsearo** [comment]
The first step would be to decide on the new boundaries for each of these and then getting (new?) badges ready for them. 

Also if we are looking into this then maybe we can even come up with some ideas for new developer tiered site awards like:
Created a set for X different consoles 
Created X total sets
Created X code notes
Created X leaderboards/X Player entries in created leaderboards (once we get authoring)
Created X rich presences (once we get authoring)
Resolved/Closed X tickets (excluding own)

Could even expand it to tiered player badges as well such as:
Master X sets
Master X sets on Y Console


Back to the current change though, these two existing awards have their own unique award type so there is probably some special code related to tracking when they are awarded and for any initial change there may need to be an update ran once for all users to properly update them to their new current tiers depending on the change. (Would need to look further into when/how these are awarded)

The code for rendering these two site awards is here and would need to be updated to include any new badges for point tiers (currently only 1 badge) and same for the awarded achievements (6 badges currently)

https://github.com/RetroAchievements/RAWeb/blob/0904de35f91e72b83eb5cb4722b6df6eb1f5f682/lib/render/user.php#L181-L205

**Comment by meleu** [comment]
@Tsearo At a first sight, I liked the suggestions. But later I noticed something that is really different from the current badges...

The current ones are mostly based on other people "appreciating" the work of the dev (of course, creating sets for popular games influence those stats).

The ones you're proposing are cool, but they're completely based on the dev action without any other form "public validation". So, I'm worried if encouraging those things with a badge can stimulate spammy/rushed actions just to get the badge.

**Comment by Tsearo** [comment]
@meleu yeah I noticed that as I was writing them up where they don't really match those two specifically but I didn't see that as being a bad thing necessarily. 
1. I don't think we should limit ourselves to only that type of dev award ("public validation")
2. They serve their purpose and so would others types (tracking dev milestones). Do extra awards detract from the existing ones?
3. The two existing ones can also be potentially abused by rushing/spamming sets for easy/free point games so that point doesn't seem very valid imo. 

Would giving extra motivation to some people to fix tickets have a bad effect on the site? Sure it could if people start making bad/rushed fixes but we already have had many successful ticket cleaning events so does that mean we should stop those?

If we are limited to only that type I'm sure there are still other suggestions that could be made that fit more in line with the "public validation", for example:
Users earned X Completed/Mastered Badges dev contributed to
X player entries in leaderboards created by dev

Again I'm just throwing out ideas for potential change/improvements since we hopefully aren't locked to past decisions as the site has changed a great deal since it was created and many artifacts still remain from that time.

**Comment by meleu** [comment]
Some of those are already being addressed by [DevQuests](https://retroachievements.org/viewtopic.php?t=13060), which also awards badges and requires QA Team's validation.

**Comment by meleu** [comment]
> If we are limited to only that type I'm sure there are still other suggestions that could be made that fit more in line with the "public validation", for example:
Users earned X Completed/Mastered Badges dev contributed to
X player entries in leaderboards created by dev

yeah, these ones are interesting indeed.

And if I'm not mistaken this kind of info is already available in that detailed [individual devstats](https://retroachievements.org/individualdevstats.php?u=meleu). So, we could (re)use the logic there to award the badges.

**Comment by Tsearo** [comment]
Ah you are right I forgot about the new DevQuest event badges for various things, that pretty much covers (or could cover) most of the ideas I mentioned 👍. 
I still think tracking overall tiered progress could be nice like [50,100,250,500,1000....] Tickets resolved for others but at least with DevQuest we can get something for solving tickets moving forward.

**Comment by Jamiras** [comment]
As the number of users grows, it becomes easier and easier to hit these targets. A newly published set for a half-recognizable game could give out a hundred unlocks and thousands of points in its first day.

The size of the userbase also largely affects the other end of the spectrum. Users who have given out a million points likely have games played by hundreds or even thousands of people.

There's currently five badges for points awarded (despite there being more than 5 tiers). 100=green, 200=pink, 300=blue, 500=silver, 800+=gold. These could be spread out more evenly across larger tiers.
* 1000=green
* 2500=green
* 5000=pink
* 10000=pink
* 25000=blue
* 50000=blue
* 100000=silver
* 250000=silver
* 500000=gold
* 1000000=gold
* 2500000=gold
* 5000000=gold
* ...

Unfortunately, the unlock badges have the numbers emblazoned into them, so we can't just recycle them at new boundaries. We'd need a new set of images (probably without the numbers). I'd propose using tiers similar to the suggested ones for points, but scaled down: 100, 250, 500, 1000, 2500, 5000, 10000, 25000, 50000, 100000, 250000, 500000, ...

**Reply by Jamiras** [reply]
Thresholds were updated in #1257.

**Comment by Jamiras** [comment]
Original request has been handled. Leaving open for ideas of other tiered developer badges. In particular, these seem reasonable:
* Number of sets created
* Masteries awarded to others
* Tickets resolved for others

The first two seem easily gameable by putting out many "free points" sets. That's still true for the badges we already have, though the sets themselves typically have fewer achievements and points, so it would be harder to gamify them.
